### PR TITLE
Fix action bar tests

### PR DESCRIPTION
**Pull request by paopaosoftwire**
Added stuff to make actions in action bar pass tests:

1. A lot of intercepts to intercept all the fetching by ShoppingList PDF.
2. Make ShoppingListPDF not async cause cypress likes it that way

- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've linted via `npm run lint`
    - can run `npm run lint_fix` to try and fix as any mistakes automatically as possible!
- [x] Make sure you've tested via `npm run test`

### CODE REVIEW

File: src/app/clients/ActionBar.cy.tsx [hunk 1]
```diff
@@ -3,122 +3,243 @@ import React from "react";
 import StyleManager from "@/app/themes";
 import { ClientsTableRow } from "@/app/clients/getClientsTableData";
 import Localization from "../localizationProvider";
```
Comment (SamYap-Softwire) [inline src/app/clients/ActionBar.cy.tsx:5]: i think this isn't from your branch, but @ import thx

File: src/app/clients/ActionBar.cy.tsx [hunk 2]
```diff
@@ -3,122 +3,243 @@ import React from "react";
...
+    interface Props {
+        selected: number[];
+    }
+
+    const MockActionBar: React.FC<Props> = ({ selected }) => {
         return (
             <Localization>
                 <StyleManager>
-                    <ActionBar data={mockData} selected={selectedIndices} />
+                    <ActionBar {...{ data: mockData, selected }} />
```
Comment (SamYap-Softwire) [inline src/app/clients/ActionBar.cy.tsx:106]: why do we need to change this?

File: src/pdf/ShoppingList/ShoppingList.tsx [hunk 3]
```diff
@@ -8,8 +8,22 @@ interface Props {
     parcelId: string;
 }
 
-const ShoppingList = async ({ text, parcelId }: Props): Promise<React.ReactElement> => {
-    const data = await prepareData(parcelId);
+const ShoppingList = ({ text, parcelId }: Props): React.ReactElement => {
+    const [data, setData] = useState<null | ShoppingListPDFDataProps>(null);
```
Comment (SamYap-Softwire) [inline src/pdf/ShoppingList/ShoppingList.tsx:12]: `<ShoppingListPDFDataProps | null>` please :)

**Review by SamYap-Softwire — APPROVED** [review]
looks good, just minor things :)

**Review by DanWilkins2107 — APPROVED** [review]

File: src/app/clients/ActionBar.cy.tsx (hunk 2 above)
Comment (paopaosoftwire) [reply src/app/clients/ActionBar.cy.tsx:106]: I wasn't the one to change this so I do not know why. I changed it back now cause I think the older version is more readable.

**Review by Var1377 — APPROVED** [review]

File: src/app/clients/ActionBar.cy.tsx [hunk 4]
```diff
@@ -2,123 +2,244 @@ import ActionBar from "@/app/clients/ActionBar";
...
+            cy.intercept(
+                "GET",
+                "https://*.supabase.co/rest/*/website_data?select=*&name=eq.lists_text",
+                {
+                    statusCode: 200,
+                    body: [{ value: "Hello worlds" }],
+                }
+            ).as("interceptFamilies");
+
+            cy.on("uncaught:exception", () => {
```
[the remark this answers is not in the document]
Comment (paopaosoftwire) [reply src/app/clients/ActionBar.cy.tsx:210]: Sorry. I was playing around with the async component failing. Removed this now.

**Review by TomCusack-Softwire — APPROVED** [review]

**Pull request merged**